source/matrix_free/task_info.cc

```diff
@@ -1060,8 +1060,13 @@ namespace internal
            ++cell)
         {
           if (!cell_vectorization_categories.empty())
-            AssertDimension(cell_vectorization_categories[cell],
-                            cell_vectorization_categories[n_active_cells]);
+            Assert(
+              cell_vectorization_categories[cell] ==
+                cell_vectorization_categories[n_active_cells],
+              ExcMessage(
+                "Currently, all ghost cells need to have the same category. "
```

Comment:
Can you add something like
```suggestion
                "Currently, all ghost cells need to have the same category, "
                "but got " + std::to_string(cell_vectorization_categories[cell]) +
                " and " + std::to_string(cell_vectorization_categories[n_active_cells)
                + "."
```
I find these kinds of assertions tremendously easier to debug if I see the value of the numbers that did not compare equal.

Reply:
@kronbichler  Done!